app/common_handlers.py

```diff
@@ -0,0 +1,464 @@
+        attributes['sub_user_journey'] = sub_user_journey
+        attributes['locale'] = locale
+        attributes['country_code'] = uprn_ai_return['response']['address']['countryCode']
+
+        return attributes
+
+    @aiohttp_jinja2.template('common-confirm-address.html')
+    async def post(self, request):
+        self.setup_request(request)
+
+        display_region = request.match_info['display_region']
+        user_journey = request.match_info['user_journey']
+        sub_user_journey = request.match_info['sub_user_journey']
+
+        if display_region == 'cy':
+            page_title = "Ydy'r cyfeiriad hwn yn gywir?"
+            locale = 'cy'
+        else:
+            page_title = 'Is this address correct?'
+            locale = 'en'
+
+        self.log_entry(request, display_region + '/' + user_journey + '/' + sub_user_journey + '/confirm-address')
+
+        session = await get_session(request)
+        attributes = await self.common_check_attributes(request, user_journey, sub_user_journey, display_region)
+
+        attributes['page_title'] = page_title
+        attributes['display_region'] = display_region
+        attributes['user_journey'] = user_journey
+        attributes['sub_user_journey'] = sub_user_journey
+        attributes['locale'] = locale
+
+        data = await request.post()
+
+        try:
+            address_confirmation = data['form-confirm-address']
+        except KeyError:
+            logger.info('address confirmation error',
+                        client_ip=request['client_ip'])
+            if display_region == 'cy':
+                flash(request, ADDRESS_CHECK_MSG_CY)
+            else:
+                flash(request, ADDRESS_CHECK_MSG)
+            return attributes
+
+        if address_confirmation == 'yes':
+
+            try:
+                if session['attributes']['countryCode'] == 'S':
+                    logger.info('address is in Scotland', client_ip=request['client_ip'])
+                    raise HTTPFound(
+                        request.app.router['CommonAddressInScotland:get'].
+                        url_for(display_region=display_region, user_journey=user_journey))
+            except KeyError:
+                logger.info('unable to check for region', client_ip=request['client_ip'])
+
+            if sub_user_journey == 'unlinked':
+                try:
+                    uprn_return = await RHService.post_unlinked_uac(request, session['case']['uacHash'],
+                                                                    session['attributes'])
+                    session['case'] = uprn_return
+                    session.changed()
+
+                    self.validate_case(uprn_return)
+
+                    raise HTTPFound(
+                        request.app.router['StartAddressHasBeenLinked:get'].url_for(display_region=display_region))
+
+                except ClientResponseError as ex:
+                    if ex.status == 404:
```

Comment:
no unit  test code coverage in this block

Reply:
I had real trouble with the unit tests for this - mocking the post errors immediately, and I couldn't work out a fix. I have tested these manually by changing the output of my RHSvc stub, so know they do trigger currently - I'll have another look.

Reply:
Unit tests added and passing